base/ltoutenc.dtx

```diff
@@ -3129,8 +3149,29 @@
 \DeclareTextSymbol{\textbaht}            \UnicodeEncodingName{"0E3F}
 \DeclareTextSymbol{\SS}                  \UnicodeEncodingName{"1E9E}
 \DeclareTextSymbol{\textcompwordmark}    \UnicodeEncodingName{"200C}
+%    \end{macrocode}
+%    
+% \changes{v2.0u}{2021/02/19}{Add \cs{textnonbreakinghyphen}, \cs{textfiguredash}
+%      and \cs{texthorizontalbar} (gh/404)}
+%    \begin{macrocode}
+\DeclareTextSymbol{\textnonbreakinghyphen} \UnicodeEncodingName{"2011}
```

Comment:
I agree that a fallback here make sense, but why not then make it simple and fall back to \mbox{-} ?

Reply:
The idea was that for a font which contains the hyphen glyph, it would be expected that \textnonbreakinghyphen is the same as the hyphen (except that it doesn't break). Of course we treat U+2010 as unbreakable anyway, so there is basically no need for that interpretation... So falling back to \mbox{m} is good.

Reply:
The OT1 definition with the hskip looks to me like an error!  It contradicts the name of the character/glyph!  

Not sure if it should be fixed now?

Reply:
Given that I introduced it about a day ago by not thinking straight, I think it should be corrected :-). Of course it should have been `\nobreak\hskip\z@`

Reply:
Aha!  OK, well done!

Reply:
Is the hskip to allow further hyphenation in the following sub-word?  Or are there other reasons?

Reply:
that is the reason, I just forgot to put the \nobreak in front.

Reply:
Although I am not suggesting that the such a skip is wrong, it can be that the local style conventions/rules sate that there should be no further break-points.  But obviously such rules vary a lot, with particular cases.

Reply:
The idea is that the definition does roughly the same that the character itself what do￼. And if you have such a style and you can't use the character without extra precaution because that would allow hyphenation afterwards.￼ so I think it does precisely what it should.

Reply:
Assuming that the character U+2011 has lccode 0, then using it will prevent hyphenation in the rest of the word.  Is that what you intend to emulate?

Reply:
Scrub that!  False memory: the skip does make it like a char with lccode 0.  Apologies!